src/tx/setCertTime.ts

```diff
@@ -262,11 +271,17 @@ export function applySetCertTimeTx(
   // deduct tx fee if certExp is not set yet or far from expiration
   /* prettier-ignore */ if (logFlags.dapp_verbose) console.log(`applySetCertTimeTx shouldChargeTxFee: ${shouldChargeTxFee}`)
 
+  let network: NetworkAccount
+  // eslint-disable-next-line security/detect-object-injection
+  if (wrappedStates[networkAccount]?.data && isNetworkAccount(wrappedStates[networkAccount].data)) {
+    // eslint-disable-next-line security/detect-object-injection
+    network = wrappedStates[networkAccount].data as NetworkAccount
+  }
   let amountSpent = bigIntToHex(BigInt(0))
   if (shouldChargeTxFee) {
     const costTxFee = scaleByStabilityFactor(
       BigInt(ShardeumFlags.constantTxFeeUsd),
-      AccountsStorage.cachedNetworkAccount
+      network
```

Comment:
is it okay to pass null here? (just want to confirm as we have a null check [above](https://github.com/shardeum/shardeum/pull/203/files#diff-801f96452344614d27069c631ad031a8fb1eca5974d2971e1acbc71bdb318c9cR165))

Reply:
this is fine because `validateSetCertTimeState` is being called beforehand and if it is `null` in there then we return in this function. Hence we never get to this point with a null option

Reply:
I should also mention that in theory the networkAccount shouldn't be able to be null there regardless. I just kept the null guards as a precaution